src/analysis/retail/mage/fire/core/CombustionActiveTime.tsx

```diff
@@ -27,48 +26,49 @@ class CombustionActiveTime extends Analyzer {
   protected abilityTracker!: AbilityTracker;
   protected combustionPreCastDelay!: CombustionPreCastDelay;
 
-  activeTime = 0;
   combustionCasts: number[] = [];
 
-  constructor(options: Options) {
-    super(options);
-    this.addEventListener(
-      Events.removebuff.by(SELECTED_PLAYER).spell(TALENTS.COMBUSTION_TALENT),
-      this.onCombustionRemoved,
-    );
-    this.addEventListener(Events.fightend, this.onFinished);
-  }
-
-  private recordCombustionEnd(event: RemoveBuffEvent | FightEndEvent) {
-    const applyBuffs = this.eventHistory.last(
-      1,
-      undefined,
-      Events.applybuff.by(SELECTED_PLAYER).spell(TALENTS.COMBUSTION_TALENT),
-    );
-    const buffApplied = applyBuffs[0]?.timestamp ?? this.owner.fight.start_time;
-    const uptime = this.filteredActiveTime.getActiveTime(buffApplied, event.timestamp);
-    this.combustionCasts[buffApplied] = uptime / (event.timestamp - buffApplied);
-    this.activeTime += uptime;
-  }
+  combustionActiveTime = () => {
+    const buffEnds = this.eventHistory.getEvents(EventType.RemoveBuff, {
+      spell: TALENTS.COMBUSTION_TALENT,
+    });
+    let activeTime = 0;
+    buffEnds.forEach((e) => {
+      const buffApplied = this.eventHistory.getEvents(EventType.ApplyBuff, {
+        spell: TALENTS.COMBUSTION_TALENT,
+        count: 1,
+        startTimestamp: e.timestamp,
+      })[0];
```

Comment:
This might throw an error in a log that doesn't have a Combustion buff; can you use `.at(0)` instead and handle the `undefined` case?